### PR TITLE
Add region to doc and handle missing region parameter

### PR DESCRIPTION
**Pull request by wimnat**

### CODE REVIEW

**Comment by wimnat** [comment]
Resolves https://github.com/ansible/ansible-modules-core/issues/868

**Comment by bcoca** [comment]
we should not be adding this to each module that needs this, it should be part of module doc fragments (guessing we need an ec2one vs aws one)

File: cloud/amazon/ec2_metric_alarm.py [hunk 1]
```diff
@@ -260,7 +265,6 @@ def main():
             insufficient_data_actions=dict(type='list'),
             ok_actions=dict(type='list'),
             state=dict(default='present', choices=['present', 'absent']),
-            region=dict(aliases=['aws_region', 'ec2_region']),
```
Comment (bcoca) [inline cloud/amazon/ec2_metric_alarm.py:262]: yes, this is redundant as we import it in ec2_argument_spec

**Comment by bcoca** [comment]
fixed via c4f64d822ce5ef9300c06433ac85a0bbfee0a75c

**Comment by bcoca** [comment]
actually only fixed docs, you can update to fix the code side

**Comment by wimnat** [comment]
All modules in one PR?

**Comment by bcoca** [comment]
not even a PR, direct commit

**Comment by wimnat** [comment]
I don't have access to direct commit.  I'll update this PR to include the fixes code side for appropriate modules.

**Comment by wimnat** [comment]
Also, i don't think the iam modules need region

**Comment by wimnat** [comment]
Done for extras - https://github.com/ansible/ansible-modules-extras/pull/1069

**Comment by wimnat** [comment]
@bcoca discarded commit for the rest of the modules.

**Pull request merged**